packages/core/src/tools/mcp-client.ts

```diff
@@ -664,6 +817,11 @@ export async function discoverTools(
               `because it has missing types in its parameter schema. Please file an ` +
               `issue with the owner of the MCP server.`,
           );
+          // Always log the schema for now to help debug
+          console.warn(
```

Comment:
This will be relatively noisy, you can guard this around config.getDebugMode() and if we are in debug mode then log this.